### ISSUE TITLE
Optimizing weights withtout gradients, is it possible?

### ISSUE THREAD

**Issue by berkankadioglu** [pytorch/pytorch#6167]
Hi, I'm sorry if this is not the right place to ask this question.

Mainly for a fully connected NN, can I write my own weight update function (I dont want to use gradients, I believe in optimizer when backpropping) according to the input/outputs and weights for each perceptron at current iteration? I am going to have the same update function for every perceptron. Can I update layers in any order I want with Pytorch?

Thank you.

**Comment by soumith** [comment]
https://discuss.pytorch.org is the right place to ask this question.
